### PR TITLE
Patch to change hardware, template, workflow CRDs to namespaced scope

### PR DESCRIPTION
**Pull request by ptrivedi**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: projects/tinkerbell/cluster-api-provider-tinkerbell/patches/0015-Make-hardware-template-workflow-CRDs-namespace-scope.patch [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+     singular: workflow
+-  scope: Cluster
++  scope: Namespaced
+   versions:
+   - name: v1alpha1
+     schema:
+diff --git a/config/default/manager_image_patch.yaml b/config/default/manager_image_patch.yaml
+index af4861a..2b3b15c 100644
+--- a/config/default/manager_image_patch.yaml
++++ b/config/default/manager_image_patch.yaml
```
Comment (pokearu) [inline projects/tinkerbell/cluster-api-provider-tinkerbell/patches/0015-Make-hardware-template-workflow-CRDs-namespace-scope.patch:59]: I think this file can be skipped. This change comes from `make release` and I dont think we need it permanently.

File: projects/tinkerbell/cluster-api-provider-tinkerbell/patches/0015-Make-hardware-template-workflow-CRDs-namespace-scope.patch [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+     spec:
+       containers:
+       # Change the value of image field below to your controller image URL
+-      - image: ghcr.io/tinkerbell/cluster-api-provider-tinkerbell:dev
++      - image: ghcr.io/tinkerbell/cluster-api-provider-tinkerbell-amd64:dev
+         name: manager
+diff --git a/config/default/manager_pull_policy.yaml b/config/default/manager_pull_policy.yaml
+index cd7ae12..74a0879 100644
+--- a/config/default/manager_pull_policy.yaml
++++ b/config/default/manager_pull_policy.yaml
```
Comment (pokearu) [inline projects/tinkerbell/cluster-api-provider-tinkerbell/patches/0015-Make-hardware-template-workflow-CRDs-namespace-scope.patch:70]: Same as above, this file can be skipped.

File: projects/tinkerbell/cluster-api-provider-tinkerbell/patches/0015-Make-hardware-template-workflow-CRDs-namespace-scope.patch [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+ 
+ 	namespacedName := types.NamespacedName{
+ 		Name: bmrc.tinkerbellMachine.Spec.HardwareName,
++		Namespace: bmrc.tinkerbellMachine.Namespace,
+ 	}
+ 
+ 	if err := bmrc.client.Get(bmrc.ctx, namespacedName, hardware); err != nil {
+@@ -176,6 +177,7 @@ func (bmrc *baseMachineReconcileContext) deprovisionHardware(hardware *tinkv1.Ha
+ 	bmc := &pbnjv1.BMC{}
+ 	namespacedName := types.NamespacedName{
```
Comment (pokearu) [inline projects/tinkerbell/cluster-api-provider-tinkerbell/patches/0015-Make-hardware-template-workflow-CRDs-namespace-scope.patch:91]: I think we need to make the BMC CRD `Namespaced` as well. 
I think the PR does not include that?

**Comment by pokearu** [comment]
Overall looks good! Some minor comments. And also we would need to make a note, that all the objects need to be on the same Namespace?

File: projects/tinkerbell/cluster-api-provider-tinkerbell/patches/0015-Make-hardware-template-workflow-CRDs-namespace-scope.patch (hunk 3 above)
Comment (ptrivedi) [reply projects/tinkerbell/cluster-api-provider-tinkerbell/patches/0015-Make-hardware-template-workflow-CRDs-namespace-scope.patch:91]: Ah, good catch. Had that in the back of my mind and forgot to do it

**Review by pokearu — APPROVED** [review]
/lgtm
/approve

**Pull request merged**